### PR TITLE
Import pyopencl.array once at the top

### LINKED ISSUE #39: Do not import PyOpenCLArrayContext into the root namespace
Move `PyOpenCLFakeNumpyNamespace` (and the linalg bit) to a submodule so that we can import `pyopencl` globally there and avoid all these repetitive, performance-killing imports
https://github.com/inducer/arraycontext/blob/322c18fe1983c6a03a7c86bf0ec3fb401422353a/arraycontext/impl/pyopencl.py#L114-L127

while still avoiding a hard dependency on pyopencl.

### PR DESCRIPTION
**Pull request by kaushikcfd**
Closes #39.

### CODE REVIEW

File: arraycontext/impl/pyopencl.py [hunk 1]
```diff
@@ -43,6 +43,12 @@
 from arraycontext.container import serialize_container, is_array_container
 from arraycontext.context import ArrayContext
 
+try:
+    import pyopencl as cl  # noqa: F401
+    import pyopencl.array as cl_array
+except ImportError:
+    pass
```
Comment (inducer) [inline arraycontext/impl/pyopencl.py:50]: Thanks for thinking about this. I don't think this suffices, unfortunately. Importing pyopencl is relatively heavyweight, and we should wait to do it until the user asks us to do so. As a result, I think the best option is to move the numpy bits to a separate file.

Comment (kaushikcfd) [reply arraycontext/impl/pyopencl.py:50]: But wouldn't we end up doing:
```
from arraycontext.impl.pyopencl.fake_numpy import maximum, ...
```
everywhere? i.e. I think we will still keep on adding a line before every math routine invocation.

Comment (inducer) [reply arraycontext/impl/pyopencl.py:50]: Not sure I understand. In my view, the only place that imports from that submodule would be the `.np` property.

Comment (kaushikcfd) [reply arraycontext/impl/pyopencl.py:50]: Aah right. I understand and I think this will work. Thanks!

**Comment by inducer** [comment]
Thanks! LGTM.

**Pull request merged**